src/editor/dag.rs

```diff
@@ -1358,6 +1406,64 @@ mod tests {
         );
     }
 
+    #[test]
+    #[ignore]
+    // This is the test cases for issue 27
+    fn level_2_undo() {
```

Comment:
Nice - I like this a lot.  Making 'regression tests' is really useful for making sure that bugs aren't accidentally reintroduced.